py/server/deephaven/table.py

```diff
@@ -1316,343 +1313,5 @@ def partition_by(self, by: Union[str, Sequence[str]], drop_keys: bool = False) -
-                return Table(j_table=j_table)
-            else:
-                return None
-        except Exception as e:
-            raise DHError(e, "unable to get constituent table.") from e
-
-    @property
-    def constituent_tables(self) -> List[Table]:
-        """Returns all the current constituent tables."""
-        return list(map(Table, self.j_partitioned_table.constituents()))
-
-    @auto_locking_op
-    def transform(self, func: Callable[[Table], Table]) -> PartitionedTable:
-        """Apply the provided function to all constituent Tables and produce a new PartitionedTable with the results
-        as its constituents, with the same data for all other columns in the underlying partitioned Table. Note that
-        if the Table underlying this PartitionedTable changes, a corresponding change will propagate to the result.
-
-        Args:
-            func (Callable[[Table], Table]: a function which takes a Table as input and returns a new Table
-
-        Returns:
-            a PartitionedTable
-
-        Raises:
-            DHError
-        """
-        try:
-            j_operator = j_unary_operator(func, dtypes.from_jtype(Table.j_object_type.jclass))
-            j_pt = self.j_partitioned_table.transform(j_operator)
-            return PartitionedTable(j_partitioned_table=j_pt)
-        except Exception as e:
-            raise DHError(e, "failed to transform the PartitionedTable.") from e
-
-    @auto_locking_op
-    def partitioned_transform(self, other: PartitionedTable, func: Callable[[Table, Table], Table]) -> PartitionedTable:
-        """Join the underlying partitioned Tables from this PartitionedTable and other on the key columns, then apply
-        the provided function to all pairs of constituent Tables with the same keys in order to produce a new
-        PartitionedTable with the results as its constituents, with the same data for all other columns in the
-        underlying partitioned Table from this.
-
-        Note that if the Tables underlying this PartitionedTable or other change, a corresponding change will propagate
-        to the result.
-
-        Args:
-            other (PartitionedTable): the other Partitioned table whose constituent tables will be passed in as the 2nd
-                argument to the provided function
-            func (Callable[[Table, Table], Table]: a function which takes two Tables as input and returns a new Table
-
-        Returns:
-            a PartitionedTable
-
-        Raises:
-            DHError
-        """
-        try:
-            j_operator = j_binary_operator(func, dtypes.from_jtype(Table.j_object_type.jclass))
-            j_pt = self.j_partitioned_table.partitionedTransform(other.j_partitioned_table, j_operator)
-            return PartitionedTable(j_partitioned_table=j_pt)
-        except Exception as e:
-            raise DHError(e, "failed to transform the PartitionedTable with another PartitionedTable.") from e
+# break circular imports
```

Comment:
I would be more ok with `table.partitioned` than with `partitioned_table`.  If remodueling is being done, it needs thought from the user side to avoid extra cognitive load.